File: Quick Python File Creater.py
from datetime import datetime
import pytz
import os
def get_time():
    global file_name
    utc_plus_8 = datetime.now(pytz.timezone('Asia/Shanghai'))
    file_name = utc_plus_8.strftime("%Y-%m-%d_%H-%M-%S")
def get_dir_path():
    global temp_path
    temp_path = os.getenv("TEMP")
def create_test_file():
    with open(temp_path+"\\"+file_name+".py", "w", encoding="utf-8") as test_file:
        test_file.write("")
    os.startfile(temp_path+"\\"+file_name+".py")
if __name__ == "__main__":
    os.system("cls")
    get_time()
    get_dir_path()
    create_test_file()

File: Quick Python File Creater.cpp
#include <iostream>
#include <fstream>
#include <string>
#include <ctime>
#include <iomanip>
#include <sstream>
#include <cstdlib>

#ifdef _WIN32
#include <Windows.h>
#endif

std::string file_name;
std::string temp_path;

void get_time() {
    // 获取当前时间并格式化为文件名
    time_t t = time(0);
    struct tm* now = localtime(&t);
    std::ostringstream oss;
    oss << std::put_time(now, "%Y-%m-%d_%H-%M-%S");
    file_name = oss.str();
}

void get_dir_path() {
    // 获取临时文件夹路径
    char* temp = std::getenv("TEMP");
    if (temp != nullptr) {
        temp_path = temp;
    }
}

void create_test_file() {
    // 创建文件并写入内容
    std::ofstream test_file(temp_path + "\\" + file_name + ".py");
    if (test_file.is_open()) {
        test_file << "";
        test_file.close();
    }
    
    // 打开文件
    std::string file_path = temp_path + "\\" + file_name + ".py";
    #ifdef _WIN32
    system((file_path).c_str());
    #endif
}

int main() {
    // 清屏
    system("cls");

    // 调用函数
    get_time();
    get_dir_path();
    create_test_file();

    return 0;
}
